### PR TITLE
Presubmit perf test

### PR DESCRIPTION
**Pull request by jdnurme**
Update presubmit to include an ~3 minute integration test. This test uses the base dataflux functions to list and download 500MB of files.

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: kokoro/presubmit.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+function run_presubmit_tests() {
+    echo Running unit tests.
+    python -m pytest dataflux_core/tests -vv --junit-xml="${KOKORO_ARTIFACTS_DIR}/unit_tests/sponge_log.xml" --log-cli-level=DEBUG
+    echo Running performance tests.
+    # -k one_shot triggers a full list and download, loading all files into memory in one shot.
+    # Alternatively, the segmented test allows us to divide the download into multiple passes
+    # to avoid OOM errors.
+    python3 -m pytest dataflux_core/performance_tests/list_and_download.py -k one_shot -vv --junit-xml="${KOKORO_ARTIFACTS_DIR}/integration_tests/sponge_log.xml" --log-cli-level=DEBUG
```
Comment (MattIrv) [inline kokoro/presubmit.sh:43]: Is there a way to set a time limit on this test so that it times out if it's running too slowly?

**Review by MattIrv — APPROVED** [review]
It looks like the test is failing right now so please fix that before submitting.

3 minutes is not bad but possibly a little bit long for a presubmit. Is there any opportunity to do a performance test that runs in something like 30 seconds, even if it's at a lower scale?

The gcsfuse performance tests doc Bernard shared the other day was interesting, I noticed that they have a suite of performance tests that you can trigger manually as a presubmit by adding a specific label when you suspect that the CL will impact performance. Maybe we should consider building out a suite like that too

File: kokoro/presubmit.sh (hunk 1 above)
Comment (jdnurme) [reply kokoro/presubmit.sh:43]: Yes, but the timeout would be added to the python test file, not the shell script.

**Comment by jdnurme** [comment]
> It looks like the test is failing right now so please fix that before submitting.
> 
> 3 minutes is not bad but possibly a little bit long for a presubmit. Is there any opportunity to do a performance test that runs in something like 30 seconds, even if it's at a lower scale?
> 
> The gcsfuse performance tests doc Bernard shared the other day was interesting, I noticed that they have a suite of performance tests that you can trigger manually as a presubmit by adding a specific label when you suspect that the CL will impact performance. Maybe we should consider building out a suite like that too

The failure is because the internal change wasn't in place when this ran. For some reason kokoro doesn't seem to want to rerun with labels if there were no changes.

**Comment by bernardhan33** [comment]
`I noticed that they have a suite of performance tests that you can trigger manually as a presubmit by adding a specific label when you suspect that the CL will impact performance. Maybe we should consider building out a suite like that too` --> yeah that would be great to have that flexibility.

`runs in something like 30 seconds, even if it's at a lower scale` -->  "+1" on this. 30 seconds sound better. But this change looks good to me.

**Review by bernardhan33 — APPROVED** [review]

**Pull request merged**